cmd/jujud/bootstrap.go

```diff
@@ -427,10 +427,37 @@ func (c *BootstrapCommand) populateTools(st *state.State, env environs.Environ)
 			SHA256:  tools.SHA256,
 		}
 		logger.Debugf("Adding tools: %v", toolsVersion)
-		if err := storage.Add(bytes.NewReader(data), metadata); err != nil {
+		if err := toolstorage.Add(bytes.NewReader(data), metadata); err != nil {
 			return errors.Trace(err)
 		}
 	}
+
+	// Store the Juju GUI archive.
```

Comment:
Wouldn't it be better to put this in a separate method (say populateGUIArchive) rather than overloading populateTools?

You could make a method (populateBinaries?) that calls them both if you wanted a single replacement for populateTools.